src/EditorFeatures/CSharpTest/Outlining/MetadataAsSource/MethodDeclarationOutlinerTests.cs

```diff
@@ -105,5 +106,24 @@ public void WithCommentsAttributesAndModifiers()
 
             AssertRegion(expectedRegion, actualRegion);
         }
+
+        [WorkItem(1174405)]
+        [Fact, Trait(Traits.Feature, Traits.Features.MetadataAsSource)]
+        public void WithNonIdentifierNames()
+        {
+            // methods from metadata could have names that don't parse as a valid C#/VB identifier
+            var tree = ParseCode(
+@"class Interop
+{
+    public Interop();
+
+    public void $Invoke();
+}");
+            var typeDecl = tree.DigToFirstTypeDeclaration();
+            var method = typeDecl.DigToFirstNodeOfType<MethodDeclarationSyntax>();
```

Comment:
Yes, it repros because `Interop()` is a constructor, not a method declaration.  I can remove the constructor for clarity if you like, I just added it because I was trying to more closely match the original repro.